alacritty_terminal/src/selection.rs

```diff
@@ -295,6 +295,14 @@ impl Selection {
         }
     }
 
+    /// Swap ends of highlighted region, returning new end of region location.
+    pub fn swap_ends(&mut self) -> Point {
+        let (start, end) = (self.region.start, self.region.end);
+        self.region.start = Anchor::new(end.point, end.side);
+        self.region.end = Anchor::new(start.point, start.side);
+        start.point
+    }
```

Comment:
when moving the cursor to the other side of a region, we also need to make the selection range go in the other direction, so we must swap our anchors.  otherwise moving the cursor from the end of the selection to the beginning will empty the selection.  Also just doing `mem::swap(&mut start, &mut end)` doesn't work.